E/Eigen/build_tarballs.jl

```diff
@@ -39,4 +39,4 @@ dependencies = Dependency[
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```

Comment:
No need to require Julia v1.6 for AnyPlatform. This package shouldn't even be instantiated by users at all, it's just header files 
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies)
```